### PR TITLE
[MIN-63] Add a spellchecker to CI and as a pre-commit hook.

### PR DESCRIPTION
**Pull request by swells2020**
This PR adds a spellchecker to both CI and the pre-commit hooks. The CI is defined in `workflows/ci.yml` and the pre-commit hook is defined in `pre-commit.yml`.

This PR also introduces a `.typos.toml` file to provide configuration for `crate-ci/typos`.

Finally, this PR corrects all typos throughout the project.

### CODE REVIEW

File: .typos.toml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[default.extend-identifiers]
+aks = "aks"
+AKS = "AKS"
+
+[default.extend-words]
+"ba" = "ba"
+
+[default]
+locale = "en-us"
```
Comment (Christopher-Norman) [inline .typos.toml]: Are we using US English for this?

Comment (swells2020) [reply .typos.toml]: I thought so, but I'll let @JonoCX make the call.

Comment (Christopher-Norman) [reply .typos.toml]: Just as we are using NHS and Mind I think it makes sense to use GB English.

Comment (JonoCX) [reply .typos.toml]: British English 👍🏻

**Review by Christopher-Norman — APPROVED** [review]
LGTM

**Review by JonoCX — APPROVED** [review]

**Pull request merged**